src/Storages/MergeTree/DataPartStorageOnDiskBase.cpp

```diff
@@ -749,8 +749,13 @@ void DataPartStorageOnDiskBase::clearDirectory(
         /// Remove each expected file in directory, then remove directory itself.
         RemoveBatchRequest request;
         for (const auto & file : names_to_remove)
-            request.emplace_back(fs::path(dir) / file);
+        {
+            if ((file.ends_with(".gin_dict") || file.ends_with(".gin_post") || file.ends_with(".gin_seg") || file.ends_with(".gin_sid"))
```

Comment:
Such an `if` with multiple `file.ends_with` conditions is not a good way to check the files.
There are already constant defined for GIN files https://github.com/ClickHouse/ClickHouse/blob/adbc819a09edbc683a8163bead927de93a354004/src/Storages/MergeTree/GinIndexStore.h#L206-L209
If you want to skip these files you should rather make a function like `isGinFile(fine_name)` in GinIndexStore.h and use those constanfs

Reply:
Thanks for reviewing the PR, the change has been made accordingly.